astroquery/eso/core.py

```diff
@@ -296,248 +218,472 @@ def _login(self, *, username: str = None, store_password: bool = False,
+            Deprecated - unused.
         """
-        if self._survey_list is None:
-            survey_list_response = self._request(
-                "GET", "http://archive.eso.org/wdb/wdb/adp/phase3_main/form",
-                cache=cache)
-            root = BeautifulSoup(survey_list_response.content, 'html5lib')
-            self._survey_list = []
-            collections_table = root.find('table', id='collections_table')
-            other_collections = root.find('select', id='collection_name_option')
-            # it is possible to have empty collections or other collections...
-            collection_elts = (collections_table.find_all('input', type='checkbox')
-                               if collections_table is not None
-                               else [])
-            other_elts = (other_collections.find_all('option')
-                          if other_collections is not None
-                          else [])
-            for element in (collection_elts + other_elts):
-                if 'value' in element.attrs:
-                    survey = element.attrs['value']
-                    if survey and survey not in self._survey_list and 'Any' not in survey:
-                        self._survey_list.append(survey)
-        return self._survey_list
-
-    def query_surveys(self, *, surveys='', cache=True,
-                      help=False, open_form=False, **kwargs):
+        _ = cache  # We're aware about disregarding the argument
+        t = _EsoNames.phase3_table
+        c = _EsoNames.phase3_surveys_column
+        query_str = f"select distinct {c} from {t}"
+        res = list(self.query_tap_service(query_str)[c].data)
+        return res
+
+    @unlimited_max_rec
+    def _print_table_help(self, table_name: str) -> None:
+        """
+        Prints the columns contained in a given table
+        """
+        help_query = (
+            f"select column_name, datatype, xtype, unit "
+            # TODO: The column description renders output unmanageable
+            # f", description "
+            f"from TAP_SCHEMA.columns "
+            f"where table_name = '{table_name}'")
+        available_cols = self.query_tap_service(help_query)
+
+        count_query = f"select count(*) from {table_name}"
+        num_records = list(self.query_tap_service(count_query)[0].values())[0]
+
+        # All this block is to print nicely...
+        # This whole function should be better written and the output
+        # shown tidier
+        nlines = len(available_cols) + 2
+        n_ = astropy.conf.max_lines
+        m_ = astropy.conf.max_width
+        astropy.conf.max_lines = nlines
+        astropy.conf.max_width = sys.maxsize
+        log.info(f"\nColumns present in the table {table_name}:\n{available_cols}\n"
+                 f"\nNumber of records present in the table {table_name}:\n{num_records}\n")
+        astropy.conf.max_lines = n_
+        astropy.conf.max_width = m_
+
+    def _query_on_allowed_values(
```

Comment:
I'm not convinced that this method is necessary, `list_columns` should deal with the help, the deprecations are handled by the deprecation decorator, the coordinates can be ducktyped (or run through a SkyCoord construction) and then the query can be called in the method that calls this one.

Reply:
You're right, and indeed, an upcoming pull request is planed to greatly improve the way the code is written. This PR focuses on switching the backend from WDB to TAP trying to impact as little as possible the API and the code structure (and it still grew larger than expected). Please be patient with these kind of apparently un-needed methods for the moment. They will disappear very soon.